code/modules/clothing/suits/hooded.dm

```diff
@@ -0,0 +1,182 @@
+// Hooded suits
+
+//Hoods for winter coats and chaplain hoodie etc
+
+/obj/item/clothing/suit/storage/hooded
+	var/obj/item/clothing/head/hood
+	var/hoodtype = null //so the chaplain hoodie or other hoodies can override this
+	var/hood_up = FALSE
+	var/toggleicon
+	action_button_name = "Toggle Hood"
+
+/obj/item/clothing/suit/storage/hooded/New()
+	toggleicon = "[initial(icon_state)]"
+	MakeHood()
+	..()
+
+/obj/item/clothing/suit/storage/hooded/Destroy()
+	qdel(hood)
+	return ..()
+
+/obj/item/clothing/suit/storage/hooded/proc/MakeHood()
+	if(!hood)
+		var/obj/item/clothing/head/hood/winter/W = new hoodtype(src)
```

Comment:
Not sure the `/winter/` is strictly necessary here. Not sure it'll do anything harmful either.